sbi/inference/snpe/snpe_base.py

```diff
@@ -240,9 +242,7 @@ def _log_prob_proposal_posterior(
     ) -> Tensor:
         raise NotImplementedError
 
-    def _run_simulations(
-        self, round_: int, num_sims: int,
-    ) -> Tuple[Tensor, Tensor, Tensor]:
+    def _run_simulations(self, round_: int, num_sims: int,) -> Tuple[Tensor, Tensor]:
```

Comment:
we dont use this function in the other methods and it kind of repeats the if-else on the rounds that happens afterwards for definiting the neural posterior. Thus, if we don't change this function, e.g., to do the merging with external data, we could as well remove it and move the if-else into the if-else in the caller code.

Reply:
Couldn't this function call `_append_to_round_bank` with the new simulations?

Reply:
Also, wouldn't it make sense to to move this into the inference base class and use it for all algorithms?